### PR TITLE
app_service: adding docker compose example to docs

### LINKED ISSUE #1633: 403 when creating container app_service
Using the newly-released v1.10.0 with the following configuration:

```
resource "azurerm_resource_group" "app" {
  name = "app-${terraform.workspace}"
  location = "West US"
}

resource "azurerm_app_service_plan" "plan" {
  name = "plan"
  location = "${azurerm_resource_group.app.location}"
  resource_group_name = "${azurerm_resource_group.app.name}"
  kind = "Linux"
  sku {
    tier = "Standard"
    size = "S1"
  }
}

resource "azurerm_app_service" "app" {
  name = "loconomics-${terraform.workspace}"
  location = "${azurerm_resource_group.app.location}"
  resource_group_name = "${azurerm_resource_group.app.name}"
  app_service_plan_id = "${azurerm_app_service_plan.plan.id}"
  site_config {
    linux_fx_version = "DOCKER|loconomics/loconomics"
  }
  app_settings {
    "WEBSITES_PORT" = "1337"
    WEBSITES_ENABLE_APP_SERVICE_STORAGE = "false"
  }
}
```

Unfortunately, every attempt to access my container throws a 403. There don't appear to be any errors when running `terraform apply`.

Is there any way to debug this further from the command line? I'm blind, Azure's web portal has some pretty huge accessibility issues, and that's why I'm trying to manage this all via Terraform. I seem to recall that viewing container/deployment logs for Linux containers may not be supported via the command line. Any other debugging tips?

Thanks.

### LINKED ISSUE #1981: creating  azurerm_app_service_plan returned an error. Status=<nil>
### Terraform Version

Terraform v0.11.8
+ provider.azurerm v1.14.0

### Affected Resource(s)

* azurerm_app_service_plan
* azurerm_app_service

### Terraform Configuration Files

```hcl

resource "azurerm_app_service_plan" "plan" {
  name                = "plan"
  location            = "${azurerm_resource_group.sandbox.location}"
  resource_group_name = "${azurerm_resource_group.sandbox.name}"
  kind                = "Linux"

  sku {
    tier = "Standard"
    size = "S1"
  }

  properties {
    reserved = true
  }
}

resource "azurerm_app_service" "app" {
  name                = "as-content-api"
  location            = "${azurerm_resource_group.sandbox.location}"
  resource_group_name = "${azurerm_resource_group.sandbox.name}"
  app_service_plan_id = "${azurerm_app_service_plan.plan.id}"

  site_config {
    linux_fx_version = "DOCKER|hello-world:latest"
  }

  app_settings {
    "WEBSITES_PORT"                       = "8080"
    "WEBSITES_ENABLE_APP_SERVICE_STORAGE" = "false"
  }
}
```
### Actual Behavior

Throw an error:

* azurerm_app_service_plan.plan: 1 error(s) occurred:

* azurerm_app_service_plan.plan: web.AppServicePlansClient#CreateOrUpdate: Failure sending request: StatusCode=0 -- Original Error: autorest/azure: Service returned an error. Status=<nil> <nil>

### Steps to Reproduce

1. `terraform apply`

### LINKED ISSUE #2827: Provision Web App for Containers
### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or "me too" comments, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

### Description

[Azure Web App for Containers](https://docs.microsoft.com/en-us/azure/app-service/containers/tutorial-multi-container-app#create-an-azure-app-service-plan) are really cool, and I think it would be awesome if this provider supported them!

As far as I can tell, the CLI [handles creating them with a couple of extra flags](https://docs.microsoft.com/en-us/cli/azure/webapp?view=azure-cli-latest#az-webapp-create). Basically, you do 
```sh
az webapp create <other required params> --deployment-container-image-name <image>
```
to create a single-container app service, and
```sh
az webapp create <other required params> --multicontainer-config-type COMPOSE --multicontainer-config-file <path-to-docker-compose>
az webapp create <other required params> --multicontainer-config-type KUBE --multicontainer-config-file <path-to-kubernetes-definition>
```
to create a multi-container app. However, I haven't been able to figure out whether - and, if it does, how - the Azure SDK for Go supports this yet.

Interestingly, if I create the service plan and web app using the CLI, according to the instructions linked above, I can then import them into my Terraform state and have the following configurations propose no changes:

```sh
az group create --name my-rg --location "West Europe"
az appservice plan create --name my-svc-plan --resource-group my-rg --sku S1 --is-linux
az webapp create --resource-group my-rg --plan my-svc-plan --name a-globally-unique-name --multicontainer-config-type compose --multicontainer-config-file a-compose-file-on-disk.yml
```
```hcl
resource "azurerm_app_service" "app" {
  name = "a-globally-unique-name"
  resource_group_name = "my-rg"
  location = "westeurope"
  app_service_plan_id = "${azurerm_app_service_plan.svcplan.id}"
}
resource "azurerm_app_service_plan" "svcplan" {
  name = "my-svc-plan"
  resource_group_name = "my-rg"
  location = "westeurope"
  sku = {
    tier = "Standard"
    size = "S1"
  }

  kind = "linux"
}
resource "azurerm_resource_group" "rg" {
  name = "my-rg"
  location = "westeurope"
}
```

### New or Affected Resource(s)

* azurerm_app_service

### PR DESCRIPTION
**Pull request by ghostinthewires**
adding docker compose examples to app service docs

Fixes #2827
Fixes #1633
Fixes #1981

### CODE REVIEW

File: website/docs/r/app_service.html.markdown [hunk 1]
```diff
@@ -172,7 +172,27 @@ The following arguments are supported:
 
 ~> **NOTE:** MySQL In App is not intended for production environments and will not scale beyond a single instance. Instead you may wish [to use Azure Database for MySQL](/docs/providers/azurerm/r/mysql_database.html).
 
-* `linux_fx_version` - (Optional) Linux App Framework and version for the AppService, e.g. `DOCKER|(golang:latest)`.
+* `linux_fx_version` - (Optional) Linux App Framework and version for the AppService, Possible values include:
+    - "DOCKER|`<image:tag>` (e.g. DOCKER|golang:latest) for single Docker container deployment.
+    - "COMPOSE|`<base64encoded_docker_compose_file>` for multiple Docker containers deployment.
+
+~> **NOTE:** When you use third party tools to update your Docker Compose definitions, you can tell Terraform not to worry about those changes with this snippet :
```
Comment (katbyte) [inline website/docs/r/app_service.html.markdown]: This might be better in the examples folder with a link to it from here.

Comment (ghostinthewires) [reply website/docs/r/app_service.html.markdown]: Yes I agree, what would you like the file to be called and how would you link it from here?

Comment (tombuildsstuff) [reply website/docs/r/app_service.html.markdown]: by the examples folder we're referring to here: https://github.com/terraform-providers/terraform-provider-azurerm/tree/master/examples which we can then link to [from the documentation](https://github.com/terraform-providers/terraform-provider-azurerm/blob/master/website/docs/r/app_service.html.markdown) as we do for example [in the VM resource](https://github.com/terraform-providers/terraform-provider-azurerm/blame/master/website/docs/r/virtual_machine.html.markdown#L17)

Would you mind pulling this example out into the `examples/` folder to give a working example of how to use this here?

**Comment by tombuildsstuff** [comment]
@ghostinthewires thanks for pushing those changes; I hope you don't mind but I've pushed a couple of commits which add some more detailed examples of using App Services to the Examples folder and then incorporated the changes from this PR; since there's quite a lot of examples I'm going to ask @katbyte to take another look.

Thanks!

**Comment by ghostinthewires** [comment]
Cool with me Tom. Thanks!

**Pull request merged**